syft/source_provider_config.go

```diff
@@ -0,0 +1,52 @@
+package syft
+
+import (
+	"crypto"
+
+	"github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft/source"
+)
+
+// SourceProviderConfig is the uber-configuration for all Syft source providers
+type SourceProviderConfig struct {
+	UserInput        string
```

Comment:
I don't think that `UserInput` belongs on this struct since it is not an option on how to get the source or optional metadata to attach. Instead it is a required parameter that cannot have a safe/sane default, so should be passed into a constructor or other function. Instead of adding a constructor (since we already have a nice default constructor) this seems to really fit as a positional argument to `SourceProviders`
```golang
func SourceProviders(imageOrDir string, cfg SourceProviderConfig) []collections.TaggedValue[source.Provider] {
   ...
}
```